### PR TITLE
nixos/kanidm: Add cacerts path to unixd service

### PR DESCRIPTION
**Pull request by Tasqa**
###### Description of changes

In order to be able to use the `kanidm_unixd` service with the `verify_ca` and `verify_hostnames` set to `true` it needs to be able to read the certificate store. This adds ssl bind paths to the unixd service file.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Can you also add a small test case to `nixos/tests/kanidm.nix` please?

**Comment by mweinelt** [comment]
> nixos/kanidm: Add cacerts path to unixd service

suggestion

> nixos/kanidm: Bind mount cacert path in unixd service

**Comment by Flakebi** [comment]
Didn’t try it but the diff looks good to me.

**Comment by Tasqa** [comment]
Updated the first commit message and added a test. First time using the tests so suggestions to improve are welcome.

File: nixos/tests/kanidm.nix [hunk 1]
```diff
@@ -44,11 +44,25 @@ import ./make-test-python.nix ({ pkgs, ... }:
         enableClient = true;
         clientSettings = {
           uri = "https://${serverDomain}";
+          verify_ca = true;
+          verify_hostnames = true;
+        };
+        enablePam = true;
+        unixSettings = {
+	  pam_allowed_login_groups = ["shell"];
```
Comment (mweinelt) [inline nixos/tests/kanidm.nix:52]: Underindented, also please leave spaces around list elemets.

File: nixos/tests/kanidm.nix [hunk 2]
```diff
@@ -44,11 +44,25 @@ import ./make-test-python.nix ({ pkgs, ... }:
...
+          home_attr = "uuid";
+          home_alias = "spn";
+          uid_attr_map = "spn";
+          gid_attr_map = "spn";
         };
       };
 
       networking.hosts."${nodes.server.config.networking.primaryIPAddress}" = [ serverDomain ];
 
+      environment.systemPackages = with pkgs; [ ripgrep ];
```
Comment (mweinelt) [inline nixos/tests/kanidm.nix:64]: Redundant, see L39.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: nixos/tests/kanidm.nix [hunk 3]
```diff
@@ -71,5 +85,8 @@ import ./make-test-python.nix ({ pkgs, ... }:
         client.wait_until_succeeds("kanidm login -D anonymous && kanidm self whoami | grep anonymous@${serverDomain}")
         rv, result = server.execute("kanidmd recover_account -c ${serverConfigFile} idm_admin 2>&1 | rg -o '[A-Za-z0-9]{48}'")
         assert rv == 0
+        client.wait_for_unit("kanidm-unixd.service")
+        rv, result = client.execute(" kanidm_unixd_status 2>&1 | rg -o 'working!'")
```
Comment (Flakebi) [inline nixos/tests/kanidm.nix:89]: 1. There’s an extra leading space in the string
2. The `2>&1` is unnecessary?
3. I see that the `-o` flag is used above as well, but I don’t understand why, it should work without
4. Last but not least, I think grep is installed by default, so a simple ` | grep 'working!'` as in line 83 should work as well :)

File: nixos/tests/kanidm.nix [hunk 4]
```diff
@@ -44,11 +44,25 @@ import ./make-test-python.nix ({ pkgs, ... }:
...
+        };
+        enablePam = true;
+        unixSettings = {
+          pam_allowed_login_groups = [ "shell" ];
+          default_shell = "/run/current-system/sw/bin/bash";
+          home_prefix = "/home/";
+          home_attr = "uuid";
+          home_alias = "spn";
+          uid_attr_map = "spn";
+          gid_attr_map = "spn";
```
Comment (Flakebi) [inline nixos/tests/kanidm.nix:58]: Are all of these needed or are they the defaults?

Comment (Tasqa) [reply nixos/tests/kanidm.nix:58]: I checked again and only `pam_allowed_login_groups` is required.

File: nixos/tests/kanidm.nix (hunk 3 above)
Comment (Tasqa) [reply nixos/tests/kanidm.nix:89]: 1. :see_no_evil: 
2. It was, only when using the `execute` command
3. & 4 regular grep :ok_hand:

File: nixos/tests/kanidm.nix [hunk 5]
```diff
@@ -71,5 +77,7 @@ import ./make-test-python.nix ({ pkgs, ... }:
         client.wait_until_succeeds("kanidm login -D anonymous && kanidm self whoami | grep anonymous@${serverDomain}")
```
Comment (Tasqa) [inline nixos/tests/kanidm.nix:77]: What is the reason this is a `wait_until_succeeds` instead of a `succeed`? In previous commands it is already established the service is up and responds right?

Comment (erictapen) [reply nixos/tests/kanidm.nix:77]: Good question, that is probably my code. I'm don't remember any reasons for it. Did you try running it with `succeed`?

Comment (Tasqa) [reply nixos/tests/kanidm.nix:77]: Yeah, no issues. Want me to change L76&77 to `succeed` in another commit?

**Comment by Tasqa** [comment]
I included the updated tests. Also is this suitable for a backport? And on an aside, if I wanted to add a `services.kanidm.package` option to override the package would that warrant a new PR?

File: nixos/tests/kanidm.nix (hunk 5 above)
Comment (erictapen) [reply nixos/tests/kanidm.nix]: Sure!

Comment (erictapen) [reply nixos/tests/kanidm.nix]: I mean there is no necessity for a second commit, just do as you prefer.

File: nixos/tests/kanidm.nix [hunk 6]
```diff
@@ -44,6 +44,12 @@ import ./make-test-python.nix ({ pkgs, ... }:
         enableClient = true;
         clientSettings = {
           uri = "https://${serverDomain}";
+          verify_ca = true;
+          verify_hostnames = true;
+        };
+        enablePam = true;
+        unixSettings = {
+          pam_allowed_login_groups = [ "shell" ];
```
Comment (erictapen) [inline nixos/tests/kanidm.nix:52]: Sorry to not have asked about this earlier, but why is this necessary? If I remove the line the test still succeeds.

Does the `shell` group have any special meaning on Unix?

**Review by erictapen — CHANGES_REQUESTED** [review]

**Review by erictapen — APPROVED** [review]
LGTM, test builds fine. I also tested, that removing the fix actually let's the new test fail :+1:

Thanks for your first contribution to Nixpkgs (afaics), @Tasqa!
